app/models/notification.rb

```diff
@@ -41,7 +41,11 @@ def email_the_user(target, actor)
   end
 
   def set_read_state( read_state )
-    self.update_attributes( :unread => !read_state )
+    if self.new_record?
```

Comment:
This method is only called on existing notifications, so this `if` shouldn't be needed.